### PR TITLE
A0-1999: Refactor github actions to use vars and secrets

### PR DESCRIPTION
**Pull request by mikogs**
# Description

This change replaces all the hardcoded values like repository names with repository vars or secrets.  These vars and secrets are set either globally in organization settings or in this repository.
In addition to that, a sub-workflow called *check-vars-and-steps* has been added.  It fails when a var or a secret is empty.  It is now added to all the workflows as a first job.

Apart from that, certain naming convention items are fixed like: invalid job names, invalid file extensions etc.

A notion page has been created with a list of all variables and secrets so we know are the actual S3 bucket names etc.

### CODE REVIEW

File: .github/workflows/deploy-feature-envs.yml [hunk 1]
```diff
@@ -293,42 +303,43 @@ jobs:
...
-      - name: GIT | Commit changes to aleph-apps repository.
+      - name: GIT | Commit changes to argocd apps repository.
         uses: EndBug/add-and-commit@v9.1.1
         env:
           APP_NAME: ${{ env.FE_APP_PREFIX }}${{ steps.get_branch.outputs.branch_appname }}
         with:
-          author_name: AlephZero Automation
-          author_email: alephzero@10clouds.com
+          author_name: ${{ vars.AUTOCOMMIT_AUTHOR }}
+          author_email: ${{ vars.AUTOCOMMIT_EMAIL }}
```
Comment (Marcin-Radecki) [inline .github/workflows/deploy-feature-envs.yml:339]: This is defined as secret in repo settings atm

File: .github/workflows/deploy-feature-envs.yml [hunk 2]
```diff
@@ -389,44 +400,45 @@ jobs:
...
         uses: EndBug/add-and-commit@v9.1.1
         env:
           IMAGE_TAG:
             ${{ env.FE_IMAGETAG_PREFIX }}${{ steps.get_branch.outputs.branch_imagetag_full }}
           APP_NAME: ${{ env.FE_APP_PREFIX }}${{ steps.get_branch.outputs.branch_appname }}
         with:
-          author_name: AlephZero Automation
-          author_email: alephzero@10clouds.com
+          author_name: ${{ vars.AUTOCOMMIT_AUTHOR }}
+          author_email: ${{ vars.AUTOCOMMIT_EMAIL }}
```
Comment (Marcin-Radecki) [inline .github/workflows/deploy-feature-envs.yml:436]: This is defined as secret in repo settings atm

File: .github/workflows/on-pull-request-commit.yml [hunk 3]
```diff
@@ -12,16 +12,23 @@ concurrency:
...
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
 
   check-excluded-packages:
+    needs: [check-vars-and-secrets]
     name: Check excluded packages
     uses: ./.github/workflows/_check-excluded-packages.yml
 
   unit-tests-and-static-checks:
+    needs: [check-vars-and-secrets]
```
Comment (Marcin-Radecki) [inline .github/workflows/on-pull-request-commit.yml:26]: This workflow does not inherit any secret, hence we don't need to wait for `check-vars-and-secrets`

File: .github/workflows/on-pull-request-commit.yml [hunk 4]
```diff
@@ -12,16 +12,23 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check-vars-and-secrets:
+    name: Check vars and secrets
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
 
   check-excluded-packages:
+    needs: [check-vars-and-secrets]
```
Comment (Marcin-Radecki) [inline .github/workflows/on-pull-request-commit.yml:21]: This workflow does not inherit any secret, hence we don't need to wait for `check-vars-and-secrets`

**Review by Marcin-Radecki — APPROVED** [review]

File: .github/workflows/_check-vars-and-secrets.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# This workflow checks if vars and secrets are present and fails if one is empty.
+# It should be included as a first step in all the workflows.
+name: Check vars and secrets
+on:
+  workflow_call:
+
+jobs:
+  main:
+    name: Check available vars and secrets
+    runs-on: self-hosted
```
[the remark this answers is not in the document]
Comment (Marcin-Radecki) [reply .github/workflows/_check-vars-and-secrets.yml]: done already

File: .github/actions/delete-feature-environment/action.yml [hunk 6]
```diff
@@ -49,10 +55,11 @@ runs:
       env:
         APP_NAME: fe-${{ steps.get-ref-properties.outputs.branch-name-for-argo }}
       run: |
-        ALEPH_PATH=$(pwd)
+        CWD_PATH=$(pwd)
```
[the remark this answers is not in the document]
Comment (mikogs) [reply .github/actions/delete-feature-environment/action.yml:58]: This is now fixed.

File: .github/workflows/deploy-feature-envs.yml [hunk 7]
```diff
@@ -293,42 +303,43 @@ jobs:
...
 
-      - name: GIT | Commit changes to aleph-apps repository.
+      - name: GIT | Commit changes to argocd apps repository.
         uses: EndBug/add-and-commit@v9.1.1
         env:
           APP_NAME: ${{ env.FE_APP_PREFIX }}${{ steps.get_branch.outputs.branch_appname }}
         with:
-          author_name: AlephZero Automation
-          author_email: alephzero@10clouds.com
+          author_name: ${{ vars.AUTOCOMMIT_AUTHOR }}
```
[the remark this answers is not in the document]
Comment (mikogs) [reply .github/workflows/deploy-feature-envs.yml]: Fixed.

File: .github/workflows/deploy-feature-envs.yml (hunk 1 above)
Comment (mikogs) [reply .github/workflows/deploy-feature-envs.yml]: Fixed.

File: .github/workflows/deploy-feature-envs.yml [hunk 8]
```diff
@@ -389,44 +400,45 @@ jobs:
...
+      - name: GIT | Commit changes to argocd apps repository.
         uses: EndBug/add-and-commit@v9.1.1
         env:
           IMAGE_TAG:
             ${{ env.FE_IMAGETAG_PREFIX }}${{ steps.get_branch.outputs.branch_imagetag_full }}
           APP_NAME: ${{ env.FE_APP_PREFIX }}${{ steps.get_branch.outputs.branch_appname }}
         with:
-          author_name: AlephZero Automation
-          author_email: alephzero@10clouds.com
+          author_name: ${{ vars.AUTOCOMMIT_AUTHOR }}
```
[the remark this answers is not in the document]
Comment (mikogs) [reply .github/workflows/deploy-feature-envs.yml]: Fixed.

File: .github/workflows/deploy-feature-envs.yml (hunk 2 above)
Comment (mikogs) [reply .github/workflows/deploy-feature-envs.yml]: Fixed.

File: .github/workflows/on-pull-request-commit.yml (hunk 4 above)
Comment (mikogs) [reply .github/workflows/on-pull-request-commit.yml]: Fixed.

**Review by Marcin-Radecki — APPROVED** [review]

File: .github/workflows/on-pull-request-commit.yml [hunk 9]
```diff
@@ -26,6 +25,12 @@ jobs:
     uses: ./.github/workflows/_build-production-node-and-runtime.yml
     secrets: inherit
```
Comment (Marcin-Radecki) [inline .github/workflows/on-pull-request-commit.yml:26]: We don't need that line anymore

File: .github/workflows/nightly-pipeline.yml [hunk 10]
```diff
@@ -10,11 +10,23 @@ concurrency:
...
+  check-vars-and-secrets:
+    name: Check vars and secrets
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
+
   build-production-node:
+    needs: [check-vars-and-secrets]
     name: Build production node and runtime artifacts (PR version)
     uses: ./.github/workflows/_build-production-node-and-runtime.yml
     secrets: inherit
```
Comment (Marcin-Radecki) [inline .github/workflows/nightly-pipeline.yml:22]: We don't need that line anymore

File: .github/workflows/on-main-branch-commit-push-liminal-node-to-ecr.yml [hunk 11]
```diff
@@ -11,7 +11,13 @@ concurrency:
...
+  check-vars-and-secrets:
+    name: Check vars and secrets
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
+
   build-liminal-node:
+    needs: [check-vars-and-secrets]
     name: Build liminal node
     uses: ./.github/workflows/_build-liminal-node.yml
     secrets: inherit
```
Comment (Marcin-Radecki) [inline .github/workflows/on-main-branch-commit-push-liminal-node-to-ecr.yml:23]: We don't need that line anymore

File: .github/workflows/on-main-branch-commit-run-button-e2e.yml [hunk 12]
```diff
@@ -11,11 +11,23 @@ concurrency:
...
+  check-vars-and-secrets:
+    name: Check vars and secrets
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
+
   build-test-node-and-runtime:
+    needs: [check-vars-and-secrets]
     name: Build test node and runtime
     uses: ./.github/workflows/_build-test-node-and-runtime.yml
     secrets: inherit
```
Comment (Marcin-Radecki) [inline .github/workflows/on-main-branch-commit-run-button-e2e.yml:23]: We don't need that line anymore

File: .github/workflows/on-main-or-release-branch-commit.yml [hunk 13]
```diff
@@ -8,11 +8,23 @@ on:
...
+  check-vars-and-secrets:
+    name: Check vars and secrets
+    uses: ./.github/workflows/_check-vars-and-secrets.yml
+    secrets: inherit
+
   build-production-node-and-runtime:
+    needs: [check-vars-and-secrets]
     name: Build production node and runtime
     uses: ./.github/workflows/_build-production-node-and-runtime.yml
     secrets: inherit
```
Comment (Marcin-Radecki) [inline .github/workflows/on-main-or-release-branch-commit.yml:20]: We don't need that line anymore

**Review by Marcin-Radecki — APPROVED** [review]

File: .github/workflows/on-main-branch-commit-run-button-e2e.yml (hunk 12 above)
Comment (mikogs) [reply .github/workflows/on-main-branch-commit-run-button-e2e.yml]: Removed.

File: .github/workflows/on-main-or-release-branch-commit.yml (hunk 13 above)
Comment (mikogs) [reply .github/workflows/on-main-or-release-branch-commit.yml]: Removed.

**Pull request merged**